Spec/RestClient.swift

```diff
@@ -824,6 +822,128 @@ class RestClient: QuickSpec {
+                        
+                        let extractHostname = { (request: NSMutableURLRequest) in
+                            NSRegularExpression.extract(request.URL!.absoluteString, pattern: "[f-j].ably-realtime.com")
+                        }
+                        let resultFallbackHosts = testHTTPExecutor.requests.flatMap(extractHostname)
+                        let expectedFallbackHosts = expectedHostOrder.map { _fallbackHosts[$0] }
+                        
+                        expect(resultFallbackHosts).to(equal(expectedFallbackHosts))
+                    }
+                    
+                    it("until httpMaxRetryCount has been reached, if an empty array of fallback hosts is provided, then fallback host functionality is disabled, use defaults instead") {
+                        let options = ARTClientOptions(key: "xxxx:xxxx")
+                        options.httpMaxRetryCount = 5
+                        options.fallbackHosts = []
+                        
+                        let client = ARTRest(options: options)
+                        client.httpExecutor = testHTTPExecutor
+                        testHTTPExecutor.http = MockHTTP(network: .HostUnreachable)
+                        testHTTPExecutor.afterRequest = { _, _ in
+                            if testHTTPExecutor.requests.count > Int(1 + options.httpMaxRetryCount) {
+                                fail("Should not retry more than \(options.httpMaxRetryCount)")
+                                testHTTPExecutor.http = nil
+                            }
+                        }
+                        let channel = client.channels.get("test")
+                        
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.publish(nil, data: "nil") { _ in
+                                done()
+                            }
+                        }
+                        
+                        expect(testHTTPExecutor.requests).to(haveCount(Int(1 + options.httpMaxRetryCount)))
+                        
+                        let extractHostname = { (request: NSMutableURLRequest) in
+                            NSRegularExpression.extract(request.URL!.absoluteString, pattern: "[a-e].ably-realtime.com")
+                        }
+                        let resultFallbackHosts = testHTTPExecutor.requests.flatMap(extractHostname)
+                        let expectedFallbackHosts = Array(expectedHostOrder.map({ ARTDefault.fallbackHosts()[$0] as! String })[0..<Int(options.httpMaxRetryCount)])
+                        
+                        expect(resultFallbackHosts).to(equal(expectedFallbackHosts))
+                    }
+                    
+                    it("until all fallback hosts have been tried, if an empty array of fallback hosts is provided, then fallback host functionality is disabled, use defaults instead") {
```

Comment:
Same problem here, it appears you are using the defaults instead of disabling the fallback functionality